test/parallel_api/iterator/iterators.pass.cpp

```diff
@@ -250,15 +257,19 @@ struct test_transform_iterator {
         test_transform_effect(in1.cbegin(), in1.cend(), in2.begin());
 
         transform_functor new_functor;
+        ref_transform_functor ref_functor;
         oneapi::dpl::transform_iterator<typename ::std::vector<T1>::iterator, transform_functor> _it1(in1.begin());
         oneapi::dpl::transform_iterator<typename ::std::vector<T1>::iterator, transform_functor> _it2(in1.begin(), new_functor);
 
         ::std::forward_list<int> f_list{1, 2, 3, 4, 5, 6};
-        oneapi::dpl::transform_iterator<decltype(f_list.begin()), transform_functor> list_it1(f_list.begin(), new_functor);
-        oneapi::dpl::transform_iterator<decltype(f_list.begin()), transform_functor> list_it2(f_list.end(), new_functor);
-        int sum = 0;
-        ::std::for_each(list_it1, list_it2, [&sum](int x){ sum += x; });
-        EXPECT_TRUE(sum == 27, "wrong result from for_each with forward_iterator wrapped with transform_iterator");
+        oneapi::dpl::transform_iterator<decltype(f_list.begin()), ref_transform_functor> list_it1(f_list.begin(), ref_functor);
+        oneapi::dpl::transform_iterator<decltype(f_list.begin()), ref_transform_functor> list_it2(f_list.end(), ref_functor);
```

Comment:
It can be simplified using `make_transform_iterator` function

Reply:
Thank you, applied.